lib/ProductOpener/Redis.pm

```diff
@@ -74,6 +117,196 @@ sub init_redis() {
 	return;
 }
 
+=head2 subscribe_to_redis_streams ()
+
+Subscribe to redis stream to be informed about user deletions.
+
+=cut
+
+sub subscribe_to_redis_streams () {
+	if (!$redis_url) {
+		# No Redis URL provided, we can't push to Redis
+		if (!$sent_warning_about_missing_redis_url) {
+			$log->warn("Redis URL not provided for streaming") if $log->is_warn();
+			$sent_warning_about_missing_redis_url = 1;
+		}
+		return;
+	}
+
+	if (!defined $redis_client) {
+		# we where deconnected, try again
+		$log->info("Trying to reconnect to Redis") if $log->is_info();
+		init_redis();
+	}
+
+	if (!defined $redis_client) {
+		$log->warn("Can't connect to Redis") if $log->is_warn();
+		return;
+	}
+
+	if (get_oidc_implementation_level() >= 4) {
+		# Read Keycloak events to process actions following user creation / deletion
+		_read_user_streams('$');
+	}
+
+	return;
+}
+
+sub _read_user_streams($search_from) {
+	# Listen for user-deleted events so that we can redact product contributions for this flavor
+	my @streams = ('BLOCK', 0, 'STREAMS', 'user-deleted');
+	# If this process handles global events then also listen for the user-registered events from Keycloak
+	if ($process_global_redis_events) {
+		push(@streams, 'user-registered');
+		push(@streams, $search_from);
+	}
+	# Note the message index to search from is provided at the end of the list in the same order as the list of keys
+	push(@streams, $search_from);
+
+	$log->info("Reading from Redis", {streams => \@streams}) if $log->is_info();
+	$redis_client->xread(
+		@streams,
+		sub {
+			my ($reply_ref, $err) = @_;
+			if ($err) {
+				$log->warn("Error reading from Redis", {error => $err}) if $log->is_warn();
+				return;
+			}
+
+			if ($reply_ref) {
+				# Process any received messages
+				my $last_processed_message_id = process_xread_stream_reply($reply_ref);
+				if ($last_processed_message_id) {
+					$search_from = $last_processed_message_id;
+				}
+			}
+
+			# Start listening for the next batch of messages
```

Comment:
Think we might need a sleep here as I noticed on opff-new that thew redis_listener service was taking 100% CPU